pandas/tseries/offsets.py

```diff
@@ -2146,8 +2167,41 @@ def onOffset(self, dt):
 
 def _tick_comp(op):
     def f(self, other):
-        return op(self.delta, other.delta)
+        if isinstance(other, Tick):
+            # Note we cannot just try/except other.delta because Tick.delta
+            # returns a Timedelta while Timedelta.delta returns an int
+            other_delta = other.delta
+        elif isinstance(other, (timedelta, np.timedelta64)):
+            other_delta = other
+        elif isinstance(other, Week) and other.weekday is None:
```

Comment:
what happens when other.weekday is not None?

Reply:
I would lump all Week in this case (use if statements *inside* if needed)

Reply:
> what happens when other.weekday is not None?

TypeError

> I would lump all Week in this case (use if statements inside if needed)

Leads to a bunch of duplicated code if we do it that way.